src/herder/TransactionQueue.cpp

```diff
@@ -473,6 +473,14 @@ TransactionQueue::tryAdd(TransactionFrameBasePtr tx)
     return res;
 }
 
+#ifdef BUILD_TESTS
+TransactionQueue::AddResult
```

Comment:
This seems to be used in just a couple of test cases. Would it be possible to cover the new interface instead?

Reply:
Yes, I agree that's better. Fixed.

Reply:
Actually I take that back. Although the `tryAdd` method is only being called in two places, one of them is in `TransactionQueueTest::add` which is then called by 100+ lines. In order to remove the test version of `tryAdd` (always passing `txRes` as arg), we need to change the `add` method and update all of its callers, which is the exact reason why we have the test-only method in the first place.

Reply:
Yes, you are absolutely right, not sure how I didn't see that. Fixed.